spatialpandas/dask.py

```diff
@@ -563,21 +563,11 @@ def _perform_get_item(self, covers_inds, overlaps_inds, x0, x1, y0, y1):
             # No partitions intersect with query region, return empty result
             return dd.from_pandas(self._obj._meta, npartitions=1)
 
-        @delayed
-        def cx_fn(df):
+        def cx_fn(df, x0, x1, y0, y1):
             return df.cx[x0:x1, y0:y1]
 
         ddf = self._obj.partitions[all_partition_inds]
-        delayed_dfs = []
-        for partition_ind, delayed_df in zip(all_partition_inds, ddf.to_delayed()):
-            if partition_ind in overlaps_inds:
-                delayed_dfs.append(
-                    cx_fn(delayed_df)
-                )
-            else:
-                delayed_dfs.append(delayed_df)
-
-        return dd.from_delayed(delayed_dfs, meta=ddf._meta, divisions=ddf.divisions)
+        return ddf.map_partitions(cx_fn, meta=ddf._meta, x0=x0, x1=x1, y0=y0, y1=y1)
```

Comment:
This might be what you're intending, but just to be explicit:

This change ignores the distinction between partitions that are fully contained in the query region (`covers_inds`) and those that overlap with the query region (`overlaps_inds`).

For those that are fully covered, the entire partition can be used as is without performing `.cx` on the partition because you're guaranteed that everything in there is inside the query region. For those that overlap, you need to perform `.cx ` on the partition because it will, in generall, contain values both inside and outside the query region.

Reply:
Thanks I'll revert this.